angular2-twitter/public/src/app/data-creator.service.ts

```diff
@@ -0,0 +1,86 @@
+import { Injectable } from '@angular/core';
+
+@Injectable()
+export class DataCreatorService {
+
+  constructor() { }
+
+  createData(sortedData: Object, colors: string[]) {
+    let i = 0;
+    let data = {
+      bar: [],
+      map: [],
+      linear: []
+    }
+    Object.entries(sortedData).forEach(([key, value]) => {
+      data.bar = this.createBarData(data.bar, key, value, colors[i]);
+      data.map = this.createMapData(data.map, key, value, colors[i]);
+      data.linear = this.createLinearData(data.linear, key, value, colors[i]);
+      i++;
+    });
+    return data;
+  }
+
+  createBarData(bar: Object[], key: string, value: any[], color: string): Object[] {
```

Comment:
jeżeli mamy więcej niż 2 argumenty w funkcji to przekazujemy je w formie obiektu